iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractAcDcConverter.java

```diff
@@ -334,5 +336,20 @@ public void remove() {
         super.remove();
     }
 
+    @Override
+    public void addDroop(AcDcConverterDroop droop) {
+        this.droopList.add(droop);
+    }
+
+    @Override
+    public List<AcDcConverterDroop> getDroopList() {
+        return droopList;
+    }
+
+    @Override
+    public AcDcConverterDroopAdder newDroop() {
+        return new AcDcConverterDroopAdderImpl(this);
+    }
```

Comment:
API users are able to mutate the content of the droop list obtained via getDroopList() which should be forbidden.

I would suggest a different approach where IIDM API defines explicitly the DroopCurve type. A DroopCurve is always defined "at once" and can never be mutated once created/added (but redefining the curve completely is possible). This would allow also checking for overlapping Umin-Umax ranges and raising exceptions as appropriate. The existing ReactiveCapabilityCurve api/impl may serve as an inspiration for this approach.